core/interfaces/i_toolbox.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The interface for a toolbox.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+'use strict';
+
+goog.provide('Blockly.IToolbox');
+
+goog.requireType('Blockly.IRegistrable');
+
+
+/**
+ * Interface for a toolbox.
+ * @extends {Blockly.IRegistrable}
```

Comment:
WorkspaceSvg makes the assumption that the toolbox is also a deletearea. Should we make it a requirement that all toolboxes be delete areas, in which case `@extends {Blockly.IDeleteArea}`

Reply:
This pr removes this assumption by adding `typeof this.toolbox_.getClientRect == 'function'`before calling it. I don't really want to require that for someone creating an external toolbox